library/LuaApi.cpp

```diff
@@ -1640,6 +1644,13 @@ static const LuaWrapper::FunctionReg dfhack_units_module[] = {
     WRAPM(Units, isKilled),
     WRAPM(Units, isGelded),
     WRAPM(Units, isDomesticated),
+    WRAPM(Units, isDemon),
+    WRAPM(Units, isTitan),
+    WRAPM(Units, isMegabeast),
+    WRAPM(Units, isGreatDanger),
+    WRAPM(Units, isSemiMegabeast),
+    WRAPM(Units, isNightCreature),
+    WRAPM(Units, isDanger),
```

Comment:
needs docs for isDemon, isTitan, isMegabeast, isSemiMegabeast, and isNightCreature